src/PrestaShopBundle/Form/Admin/Catalog/Category/AbstractCategoryType.php

```diff
@@ -125,12 +126,22 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'type' => TextWithLengthCounterType::class,
                 'required' => false,
                 'options' => [
-                    'max_length' => 255,
+                    'max_length' => 70,
```

Comment:
@matks I know we talked about this earlier, but I don't really like these hardcoded values. :/

Reply:
We must use const for these kinds of values, at least to understand why 70 and not 255 :thinking:

Reply:
That's what I want to do, but do you have any suggestion where they should be defined? For me it seems that these values should be defined somewhere in our `Domain`.

Reply:
If you wan to put const in Domain, it should be in valueObject;

Reply:
Yeah, that makes sense, but again, should it be `MetaTitle` VO? Should it be in `Domain/ValueObject` or `Domain/Category/ValueObject`?

Reply:
ping @matks

Reply:
If the chars limit is related to the data (it's either linked to how it's stored or linked to the meaning of this data) we could put it in the related entity. Thats means the ObjectModel :trollface: .

If the chars limit is linked to the form or the display, it's in the right place, in the CategoryType. We can add a `const` for it but if it's used once I think we're fine. Adding a const used only once feels wrong to me, like having
```
->add('menu_thumbnail_images', FileType::class, [
                'multiple' => true,
                'required' => false,
            ])
```
become
```
->add(self:MENU_THUMBNAIL_IMAGES, FileType::class, [
                'multiple' => true,
                'required' => false,
            ])
```

Reply:
You don't have any problem to do it with variables :trollface: 
But anyway, if it used twice use const, if the value is weird like moving from 255 to 70 or using 88, could be cool to add at least a comment or a const.

Reply:
We need to decide where it should be defined. :) I think that `MetaTitle` VO could be most appropriate place for it, however, it seems a bit redundant to have VO for that.

Reply:
It would be strange to create an unused VO juste to carry this constant 🤔.

I would feel better with a class that makes it unambiguous it's only there to carry constants. Like MetaOptimization.php or SEOConfiguration.php

Reply:
> Like MetaOptimization.php or SEOConfiguration.php

The idea is good enough I think, but naming is misleading imo. Image we may have (and I think we do) constraints for non SEO related fields.

Reply:
It's not a constraint, it's rather a setting.
`MetaSettings` 😄 ?

Reply:
Is it really a setting? For me it's more like length constraint. Same as we have `int` constraint for IDs.

Reply:
It looks like a Setting or a Property 🤔.

Classic database character types have maximum length of 255. 255 is not a constraint: it is a property of these types, and we enforce content that go into these fields to be less than 255 characters because of this property.

On the other hand, SEO friendly URLs must only contain alphanumeric chars or - . No space, no _ are allowed. This is a constraint. If tomorrow we decide that passwords cannot contain white space, this will be a constraint, not a property as a password can have white spaces in it.

That's how I see it 🤝 wdyt ?

Reply:
Both `Length` and `alphanumeric` are constraints for particular value imo. Value of length constraint is 70 and has nothing to do with database (or at least it shouldn't have). 

In the end, I think we can settle with either your suggestion `MetaSettings` or mine `MetaConstraints`. ^^

Reply:
MetaSettings for the name with the phpDoc `Constraints for meta [...]` ? ❤️

Reply:
I've introduced `SeoSettings` class instead, because for me it felt like `MetaSettings` could be confused with `Meta` subdomain.